specs/interop/derivation.md

```diff
@@ -132,19 +133,31 @@ is a deposit transaction sent to the zero address.
 
 This transaction MUST have the following values:
 
-1. `from` is `0xdeaddeaddeaddeaddeaddeaddeaddeaddead0002` (the address of the
+1. `from` is `0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001` (the address of the
    [L1 Attributes depositor account](../protocol/deposits.md#l1-attributes-depositor-account))
 2. `to` is `0x0000000000000000000000000000000000000000` (the zero address as no EVM code execution is expected).
 3. `mint` is `0`
 4. `value` is `0`
 5. `gasLimit` is set `36000` gas, to cover intrinsic costs, processing costs, and margin for change.
 6. `isSystemTx` is set to `false`.
-7. `data` is the preimage of the [L2 output root](../glossary.md#l2-output-root-proposals)
+7. `data` is the preimage of the [L2 output root]
    of the replaced block. i.e. `version_byte || payload` without applying the `keccak256` hashing.
+8. `sourceHash` is computed with a new deposit source-hash domain, see below.
 
 This system-initiated transaction for L1 attributes is not charged any ETH for its allocated
 `gasLimit`, as it is considered part of state-transition processing.
 
+[L2 output root]: ../glossary.md#l2-output-root-proposals
+
+#### Optimistic Block Source-hash
+
+The source hash is [computed](../protocol/deposits.md#source-hash-computation)
+with a source-hash domain: `4` (instead of `1`),
```

Comment:
noob question, but why keep making new deposit source domains?

Interop does already introduce one more @ `3` for the special deposit that closes the EVM-aware deposit window in the `L1Block` contract. Is it useful to introduce another one here, rather than create a more generic system tx source domain w/ `3` (or repurpose `2`, currently for upgrades, collapsing `3` + `4` into it, and just calling it `System`)?

What actually uses these identifiers to delineate between deposit sources?

Reply:
Also, if we're changing the domain then it's no longer an L1 attributes deposited tx based on the [rules for computing the source hash is computed](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/deposits.md#source-hash-computation). You'd have to introduce a new tx type to avoid complicating the spec.

Reply:
@Inphi It's not meant to be a L1 attributes deposited tx, it's unrelated to L1.

Reply:
@clabby it's a safety thing, to ensure two types of system txs can never accidentally have the same source-hash. It may be a bit overkill, but it's very easy to spec / implement the idea of a unique domain number to combine with it.

Reply:
No tool / thing currently needs it from a non-safety use-case, but like the output-root, if we ever need to unpack the preimage of an unknown type of source-hash, then we can do so safely.

Reply:
In which case, let's fix the blurb on L131 regarding the L1 deposited tx.